openssl/src/symm.rs

```diff
@@ -387,6 +401,41 @@ impl Crypter {
         }
     }
 
+    /// Sets the length of the authentication tag to generate in AES CCM.
+    ///
+    /// When encrypting with AES CCM, the tag length needs to be explicitly set in order
+    /// to use a value different than the default 12 bytes.
+    pub fn set_tag_len(&mut self, tag: &[u8]) -> Result<(), ErrorStack> {
+        unsafe {
+            assert!(tag.len() <= c_int::max_value() as usize);
+            // NB: this constant is actually more general than just GCM.
+            cvt(ffi::EVP_CIPHER_CTX_ctrl(
+                self.ctx,
+                ffi::EVP_CTRL_GCM_SET_TAG,
+                tag.len() as c_int,
+                ptr::null_mut(),
+            )).map(|_| ())
+        }
+    }
+
+    /// Feeds total plaintext length to the cipher.
+    ///
+    /// The total plaintext or ciphertext length MUST be passed to the cipher when it operates in
+    /// CCM mode.
+    pub fn ccm_update(&mut self, input: &[u8])-> Result<(), ErrorStack> {
```

Comment:
Can we name this something like `set_data_len` and maybe just take a usize rather than the full buffer? It'd make it more clear that it's only using the length and not the data.

Reply:
Sure. Should I update the signature for `set_tag_len` as well?